### PR TITLE
Add a Redis cache to the REST API

### LINKED ISSUE #8000: Cache list transactions
### Problem

The list transactions endpoint is the most heavily used endpoint and is also one of the slowest queries. `/api/v1/accounts/{id}` has similar problems.

### Solution

For both `/api/v1/transactions` and `/api/v1/accounts/{id}`:

* Query Redis by list of transaction timestamps and get JSON transaction output
* Update Redis on cache miss

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add a Redis cache to `/api/v1/transactions`
* Add Redis configuration to REST chart
* Remove temporary snyk Gradle plugin workaround

**Related issue(s)**:

Fixes #8000

**Notes for reviewer**:

### Before

| URL | VUS | Pass% | Pass RPS | Avg. Req Duration |
|----------|-----|-------|----------|-------------------|
| /transactions?limit=100 | 1500 | 100.00 | 619.50/s | 2392.61ms |
| /transactions?account.id=1833905&limit=100 | 1500 | 100.00 | 575.79/s | 2575.99ms |
| /transactions?transactionType=CRYPTOTRANSFER&limit=100&order=asc | 1500 | 100.00 | 552.42/s | 2676.92ms |

### After

| URL | VUS | Pass% | Pass RPS | Avg. Req Duration |
|----------|-----|-----|----------|-------------------|
| /transactions?limit=100 | 1500 | 100.00 | 942.86/s | 1556.75ms |
| /transactions?account.id=1833905&limit=100 | 1500 | 100.00 | 869.74/s | 1693.94ms |
| /transactions?transactionType=CRYPTOTRANSFER&limit=100&order=asc | 1500 | 100.00 | 769.37/s | 1909.90ms |

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/transactions.js [hunk 1]
```diff
@@ -248,13 +251,13 @@ const getStakingRewardTransferList = async (stakingRewardTimestamps) => {
 
   const positions = _.range(1, stakingRewardTimestamps.length + 1).map((position) => `$${position}`);
   const query = `
```
Comment (jascks) [inline hedera-mirror-rest/transactions.js:253]: It looks like the SQL changes are just reformatting performed by Spotless or something else and were not materially change to accommodate the cache for some reason, right?

File: hedera-mirror-rest/cache.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    this.redis.on('connect', () => logger.info(`Connected to ${uriSanitized}`));
+    this.redis.on('error', (err) => logger.error(`Error connecting to ${uriSanitized}: ${err.message}`));
+    this.redis.on('ready', () => {
+      this.#setConfig('maxmemory', config?.redis?.maxMemory);
+      this.#setConfig('maxmemory-policy', config?.redis?.maxMemoryPolicy);
+      this.ready = true;
+    });
+  }
+
+  #setConfig(key, value) {
```
Comment (jascks) [inline hedera-mirror-rest/cache.js:58]: Did not get anything on a quick search. Why the `#` prefix? Just curious...

File: hedera-mirror-rest/transactions.js (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/transactions.js:253]: Right, my intellij formatted them and spotless apparently doesn't understand SQL within strings so kept it.

File: hedera-mirror-rest/cache.js (hunk 2 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/cache.js:58]: It's the ES2020 private methods syntax

**Review by jascks — DISMISSED** [review]
LGTM

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by mgoelswirlds — DISMISSED** [review]
LGTM

**Review by edwin-greene — DISMISSED** [review]
Looks good

**Review by xin-hedera — DISMISSED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**